app/src/src/screens/menu/EditProfileScreen.jsx

```diff
@@ -12,12 +20,56 @@ class EditProfileScreen extends Component {
   componentDidMount = async () => {
     const userData = await getUserData();
     this.setState({ user: userData });
+    /* await this.getJobs();
+    // repeat if there's more pages
+    while (this.state.pages) await this.getJobs();
+    this.setState({
+      jobs: this.sortJobs(),
+    });
+    this.sortJobs();
+    this.setState({ done: true }); */
+  };
+
+  sortJobs = () => {
+    const nulls = this.state.jobs.filter((job) => job.end_date === null);
+    const sortedNulls = nulls.sort(
+      (a, b) => new Date(a.start_date) - new Date(b.start_date)
+    );
+    const notNulls = this.state.jobs.filter((job) => job.end_date !== null);
+    const sortedNotNulls = notNulls.sort(
+      (a, b) => new Date(a.end_date) - new Date(b.end_date)
+    );
+    return [...sortedNulls, ...sortedNotNulls];
+  };
+
+  getJobs = async () => {
+    /* TODO: for some reason, the filtering by creator doesn't filter by creator (it can filter by job_type though ? )
+       currently, it returns all the jobs for all the users */
```

Comment:
Yeah our jobs endpoint isn't fully functional yet. The creator filter doesn't exist on it right now. The syntax you have is correct for when it is implemented next term.